forwarding/test_match_host_forwarded_regex.py

```diff
@@ -0,0 +1,538 @@
+    backends = [
+    ]
+
+    tempesta = {
+    }
+
+    requests_opt = [
+    ]
+
+    blocked_response_status = "403"
+    success_response_status = "200"
+    chains = []
+
+    def add_client(self, cid):
+        class_name = type(self).__name__
+        if class_name.find("Http") != -1:
+            client = {"id": cid, "type": "deproxy", "addr": "${tempesta_ip}", "port": "80"}
+        else:
+            client = {"id": cid, "type": "deproxy_h2", "addr": "${tempesta_ip}", "port": "443", "ssl": True}
+        self.clients.append(client)
+
+    def init_chain(self, req_opt):
+        ch = chains.base(uri=req_opt["uri"])
+        if req_opt["block"]:
+            for header, value in req_opt["headers"]:
+                ch.request.headers.delete_all(header)
+                ch.request.headers.add(header, value)
+                ch.request.update()
+                ch.fwd_request = None
+        else:
+            for request in [ch.request, ch.fwd_request]:
+                for header, value in req_opt["headers"]:
+                    request.headers.delete_all(header)
+                    request.headers.add(header, value)
+                    request.update()    
+        self.chains.append(ch)
+
+    def setUp(self):
+        del self.chains[:]
+        count = len(self.requests_opt)
+        for i in range(count):
+            self.init_chain(self.requests_opt[i])
+            self.add_client(i)
+        tester.TempestaTest.setUp(self)
+
+    def start_all(self):
+        self.start_all_servers()
+        self.start_tempesta()
+        time.sleep(2) #It is necessary to wait until regexes are compiled
+        self.start_all_clients()
+        self.deproxy_manager.start()
+        self.assertTrue(self.wait_all_connections())
+
+    def process(self, client, chain, request, sid):
+        if isinstance(client, deproxy_client.DeproxyClientH2):
+            client.make_request(request["h2_headers"]) 
+        else:
+            client.make_request(chain.request.msg)      
+            
+        client.wait_for_response()
+        if chain.fwd_request:
+            last_response_status = client.last_response.status    
+            self.assertEqual(self.success_response_status, last_response_status)
+            last_body = client.last_response.body
+            self.assertEqual(str(sid), last_body)
+        else:
+            last_response_status = client.last_response.status
+            self.assertEqual(self.blocked_response_status, last_response_status)
+
+    def test_chains(self):
+        """
+        Send requests with different URI and headers
+        and check correctness of forwarding
+        by compare last response body with sid of
+        whaiting server.
+        """
+        self.start_all()
+        count = len(self.chains)
+        for i in range(count):
+            sid = self.requests_opt[i]["sid"]
+            client = self.get_client(i)
+            self.process(client, self.chains[i], self.requests_opt[i], sid)
```

Comment:
I suggest use `parametrize` here
```suggestion
    @parameterize.expand(
        [
            param(
                name="host_testwiki_uri_testwiki",
                uri="/testwiki",
                host="testwiki.com",
                block=False,
                sid=1
            )
        ]
    )
    def test(self, name, uri, host, block, sid):
        """
        Send requests with different URI and headers
        and check correctness of forwarding
        by compare last response body with sid of
        whaiting server.
        """
        self.start_all()
        client = self.get_client("deproxy")
        client.restart()
        request = client.create_request(method="GET", uri=uri, authority=host, headers=[])
        client.send_request(request)

        if not block:
            last_response_status = client.last_response.status
            self.assertEqual(self.success_response_status, last_response_status)
            last_body = client.last_response.body
            self.assertEqual(str(sid), last_body)
        else:
            last_response_status = client.last_response.status
            self.assertEqual(self.blocked_response_status, last_response_status)
```
This will make the test more reliable. Or you should use `self.subTest` construction here. https://github.com/tempesta-tech/tempesta-test/blob/ab3c689c47416961f7291dcd972d4354046c78ca/selftests/test_port_checks.py#L36